lib/SILOptimizer/Mandatory/DataflowDiagnostics.cpp

```diff
@@ -125,10 +165,12 @@ class EmitDFDiagnostics : public SILFunctionTransform {
       return;
 
     SILModule &M = getFunction()->getModule();
+    ConstExprEvaluator constantEvaluator(M);
```

Comment:
@marcrasi  I wonder whether it would be better to create `ConstExprEvaluator` and invoke `diagnosePoundAssert` only when the experimental -enable-static-assert option is enabled.

Reply:
yes, that sounds good. will do.